pkg/obfuscator/regex.go

```diff
@@ -0,0 +1,48 @@
+package obfuscator
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+type regexObfuscator struct {
+	ReplacementReporter
+	pattern  *regexp.Regexp
+	location string
+}
+
+func (r *regexObfuscator) FileName(s string) string {
+	if r.location == "All" || r.location == "FileName" {
+		return r.replace(s)
+	}
+	return s
+}
+
+func (r *regexObfuscator) Contents(s string) string {
+	if r.location == "All" || r.location == "FileContents" {
```

Comment:
use the constants from the schema please: ObfuscateTargetAll and ObfuscateTargetFileContents do exist

Reply:
same for the FileName above

Reply:
Leaking abstractions?

Reply:
what are you leaking from where? that's why you have a schema, so you can share it

Reply:
Updated